### ISSUE TITLE
Color emoji not rendered

### ISSUE THREAD

**Issue by smaznet** [lovell/sharp#2506]
Are you using the latest version? Is the version currently in use as reported by `npm ls sharp` the same as the latest version as reported by `npm view sharp dist-tags.latest`?

Yes sharp@0.27.0 

What are the steps to reproduce?
 i tried this code to convert svg to text
 
 ``` xml
 <svg height="117" width="512">
  <rect x="0" y="0" width="512" height="117" stroke-width="3px" stroke="white" fill="black" rx="30"/>
  <text x="50%" y="50%" fill="white" font-size="5em" baseline-shift="-25%" lengthAdjust="spacing" textLength="90%" text-anchor="middle">😂</text>
</svg>
```
i tested it on w3schools.com it shows this
![image](https://user-images.githubusercontent.com/10239885/103351033-211dbe00-4ab7-11eb-93ed-86498dc83f15.png)

but using sharp i got this 
![image](https://user-images.githubusercontent.com/10239885/103351041-2975f900-4ab7-11eb-969c-6746fb7cc0da.png)

and i installed noto-color-emoji

**Comment by lovell** [comment]
Hi, this is probably due to an issue with rendering coloured fonts, most likely in cairo as there are a few possible candidates here - https://gitlab.freedesktop.org/cairo/cairo/-/issues?scope=all&utf8=%E2%9C%93&state=opened&search=emoji

**Comment by vasanthdeveloper** [comment]
Will this issue ever get fixed 🤔 the upstream issue has been open from 2 years now 🤷‍♂️

**Comment by lovell** [comment]
This is likely to be due to https://gitlab.freedesktop.org/cairo/cairo/-/issues/326

**Comment by PeterKohlichhaus** [comment]
I'm having the same issue and I haven't been able to solve it. I would like to use Twitter Color Emoji rather than noto though. Creating a font.conf with hinting enabled did nothing, and FC_DEBUG=1024 does say the file is being loaded. FC_DEBUG=1 shows a bunch of fonts from the <dir> path, but all of them have hinting set to false.
pango-view doesn't seem to help either. while sharp can render emojis with outline (but no color), pango-view shows no emojis at all.

**Comment by penchochris** [comment]
Any solution to this @lovell ?

**Comment by andre-luiz1997** [comment]
I am getting the same issue on sharp. I have tested with noto-color-emoji and with emoji-one on ubuntu 22, but without success. 

I have already tried 

> This is likely to be due to https://gitlab.freedesktop.org/cairo/cairo/-/issues/326

**Comment by Runi-c** [comment]
Same issue, in my case `pango-view` does render color emoji but `sharp` does not. Have spent multiple hours playing with fontconfig configs and debug vars, as well as different fonts.

The fact that I can get `pango-view` to render what I want perfectly but not `sharp` seems to suggest there's a possible bug in the way `sharp` or perhaps `libvips` is using `pango`, right? I don't understand how these components connect well enough to figure out how it's possible that only `sharp` isn't working...

**Comment by kleisauke** [comment]
This is due to that librsvg renders all text as paths to workaround a Pango bug, see for details:
https://gnome.pages.gitlab.gnome.org/librsvg/devel-docs/text_layout.html#emoji-is-broken

So, this issue is currently blocked on https://gitlab.gnome.org/GNOME/librsvg/-/issues/911.

Note that the text operation can be used as an alternative, which should work with colored emoji glyphs after PR https://github.com/lovell/sharp-libvips/pull/228 (sharp v0.33.5).

**Comment by rion18** [comment]
@kleisauke Hi! I'm having this same issue, however, you mentioned

> Note that the text operation can be used as an alternative, which should work with colored emoji glyphs after PR https://github.com/lovell/sharp-libvips/pull/228 (sharp v0.33.5).

What text operation can we use as an alternative that would work with colored emoji? 

It seems this 
```ts
sharp({ 
  text: { 
    text: '😀😀😀', 
    font: "sans-serif", 
    rgba: true 
  } 
});
```
still doesn't work...

**Comment by lovell** [comment]
@rion18 This issue relates to SVG rendering. If you're still having problems with the existing text rendering feature of sharp, please open a new issue and provide a complete, minimal code sample that allows someone else to reproduce.

**Comment by rion18** [comment]
Just wanted some clarification on the Pango bug workaround and then after some trial and error wanted to add to what @kleisauke mentioned. Using 

```ts
sharp({ 
  text: { 
    text: '😀😀😀', 
    font: "sans-serif", 
    rgba: true 
  } 
});
```

only works if fontconfig is setup correctly with colored emoji fonts or a font with an emoji font fallback. The recommendation did work! Sorry for the confusion @lovell !
